plugins/notion/src/App.tsx

```diff
@@ -32,6 +36,12 @@ export function AuthenticatedApp({ context }: AppProps) {
         })
     }, [databaseConfig])
 
+    useEffect(() => {
+        if (!databaseConfig) return
+
+        context.databaseIdMap.set(databaseConfig.id, context.collection.id)
+    }, [context, databaseConfig])
```

Comment:
Why is this needed?

Reply:
<img width="353" alt="image" src="https://github.com/user-attachments/assets/534f380d-b47c-4461-be95-87efd450539b">

It maps the selected Notion database to the current collection id. Could be better if moved to the `onSubmit`.

Reply:
Doesn't this

https://github.com/framer/plugins/pull/120/files/ffde46269b1e791e784d2afcf9650aeece4748b4#diff-0b64b774d3bc268fbeb7b15c667dac368a9c45ee00413a89778ec0efa2d4ef1fR784-R786

also do that, though?

Reply:
Ah, this would work only if `databaseId` is set which is not the case when importing a new one (importing vs syncing).

But you are right in the sense that we should never enter this if statement because when we're importing a new database, `databaseId` will be `null`, and when we're syncing an existing database, the ID will already be in the map from the collection iteration. So this is effectively dead code that can never be reached in either scenario.

Reply:
OK - best to remove then, it sounds like